mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -472,11 +472,15 @@ fn unmount(mount_point: &Path) {
         Ok(result.success())
     }
 
-    // Try both FUSE 2 and FUSE 3 versions, since we don't know where we're running
-    for bin in ["fusermount", "fusermount3"] {
-        if matches!(run_fusermount(bin, mount_point), Ok(true)) {
-            return;
+    // Loop a bit to give any slow/async FUSE requests time to finish
+    for i in 1..4 {
+        // Try both FUSE 2 and FUSE 3 versions, since we don't know where we're running
+        for bin in ["fusermount", "fusermount3"] {
+            if matches!(run_fusermount(bin, mount_point), Ok(true)) {
+                return;
+            }
         }
+        std::thread::sleep(i * Duration::from_secs(1));
```

Comment:
Does this help fix occasional tests failing to unmount?

I was getting this error consistently last week.

```
---- fuse_tests::fork_test::run_in_foreground stdout ----
thread 'fuse_tests::fork_test::run_in_foreground' panicked at mountpoint-s3/tests/fuse_tests/fork_test.rs:482:5:
failed to unmount
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
Also curious about this: I think @vladem was getting similar errors initially in #715.

Reply:
yep, that should fix the errors I observed; I was thinking about a change that targets specifically "retriable" errors, but your approach also lgtm:
```
fn run_fusermount(bin: &str, mount_point: &Path) -> Result<bool, Box<dyn std::error::Error>> {
    let st = Instant::now();
    loop {
        if st.elapsed() > MAX_WAIT_DURATION {
            panic!("unmount is taking too long")
        }
        let child = Command::new(bin).stderr(Stdio::piped()).arg("-u").arg(mount_point).spawn()?;
        let result = child.wait_with_output()?;
        let stderr_str = String::from_utf8(result.stderr).expect("mountpoint log is not a valid UTF-8");
        // retry a specific error otherwise return immediately
        if result.status.success() || !stderr_str.contains("Device or resource busy") {
            return Ok(result.status.success())
        }
        std::thread::sleep(Duration::from_millis(100));
    }
}
```

Reply:
Yeah. Vlad's approach is probably smarter, I'm just lazy and wanted to go to bed :-)